airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/state/SourceStateIteratorProcessor.java

```diff
@@ -0,0 +1,30 @@
+package io.airbyte.cdk.integrations.source.relationaldb.state;
+
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.AirbyteMessage.Type;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import java.time.Instant;
+
+public interface SourceStateIteratorProcessor<T> {
```

Comment:
Looking at the implementation of this is `MySqlInitialSyncStateIteratorProcessor` -  this seems to be a wrapper around StateManager? Perhaps we should rename this to SourceStateManager?

A more stretch goal is to see if there is a way to combine parts of this code and the various StateManagers since there is definitely some duplicate code (move certain fields to generics)

Reply:
Discussed offline and we decide to keep this as is- 

I didn’t make the stateManager change, because:
- Abstracting all logic from various statemanager is difficult because each has different function signature to generate state message
- Merging Processor logic into stateManager within the same connector is problematic; because then we would have to create an abstract class of the stateManager on top of global and PerStream so we don’t duplicate the Processor logic in it, and since we are creating another layer anyway we would prefer composition than inheritance.